File: src/main/java/com/dh/ondot/schedule/core/exception/UnavailableOpenAiServerException.java
package com.dh.ondot.schedule.core.exception;

import com.dh.ondot.core.exception.BadGatewayException;

import static com.dh.ondot.core.exception.ErrorCode.UNAVAILABLE_OPEN_AI_SERVER;

public class UnavailableOpenAiServerException extends BadGatewayException {
    public UnavailableOpenAiServerException() {
        super(UNAVAILABLE_OPEN_AI_SERVER.getMessage());
    }

    @Override
    public String getErrorCode() {
        return UNAVAILABLE_OPEN_AI_SERVER.name();
    }
}

File: src/main/kotlin/com/dh/ondot/schedule/core/exception/UnavailableOpenAiServerException.kt
package com.dh.ondot.schedule.core.exception

import com.dh.ondot.core.exception.BadGatewayException
import com.dh.ondot.core.exception.ErrorCode.UNAVAILABLE_OPEN_AI_SERVER

class UnavailableOpenAiServerException :
    BadGatewayException(UNAVAILABLE_OPEN_AI_SERVER.message) {
    override val errorCode: String get() = UNAVAILABLE_OPEN_AI_SERVER.name
}
